Module 2.1

N1ckky opened the pull request:
Send first module for review

N1ckky committed:
Added module 2.1

Pushed task from module 2.1

N1ckky committed:
clean code sonar

Check code using sonar plugin, delete some lines

AndreiValnianka commented on Nick_Maven/ErrorExeptions/Main.java:
```diff
@@ -0,0 +1,141 @@
+
+
+        //    Посчитать средний балл по всем предметам студента
+        int fooGrade = -666;
```
Please avoid 'magic numbers'.  You should create constants.

AndreiValnianka commented on Nick_Maven/ErrorExeptions/Main.java:
```diff
@@ -0,0 +1,141 @@
+            for (int j = 0; j < student.subject.length; j++) {
+                if (student.subject[j].getNameSubject().equals(stringForCheckIfStudendHaveUnnamedField) ||
+                        student.subject[j].getGrade() == fooGrade) {
+                    continue INNER;
```
Not a good practice in Java

AndreiValnianka commented on Nick_Maven/ErrorExeptions/ProjectPackage/Student.java:
```diff
@@ -0,0 +1,132 @@
+        }
+    }
+
+    class FacultyNameException extends Exception {
```
Create a separate package for exceptions and put all exceptions to this package as divided classes

AndreiValnianka reviewed (changes requested):
see comments

N1ckky committed:
Error correction

1) avoid 'magic numbers';
2) loop labels removed;
3) create the separate package for exceptions and put all exceptions to this package as divided classes
